docs(msp): update base protocol specifications (#12)
stryker-mutator/editor-plugins: packages/mutation-server-protocol/README.md

> [!NOTE]  
> Inspired by the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/overviews/lsp/overview/)

This document describes the mutation server protocol.

## Base Protocol

The base protocol exchanges [JSON-RPC 2.0](https://www.jsonrpc.org/) messages between the client and the server via a socket connection. The server must answer each request from the client with a response. The server may also send notifications to the client. The protocol is designed to be language agnostic and can be used with any programming language.

The mutation server must:

1. Open a socket to accept incoming client connections.

> 2. Use the port specified by the client via the `--port <port_number>` argument, if provided. If no port is specified, the server must automatically select an available port.

Reviewer: ```suggestion
2. Provide a method such as command argument to configure  a static port. If a static port number is provided it must be used. If the static port is already in use the server must exist with an error. If no port number is specified the server must automatically select an available port.
```

> 2. Provide a method such as command argument to configure a static port number. If a static port number is provided it must be used. If the static port number is already in use the server must exist with an error. If no port number is specified the server must automatically select an available port.

Reviewer: ```suggestion
2. Provide a method such as command argument to configure a static port number. If a static port number is provided it must be used. If the static port number is already in use the server must exist with an error. If a port number is not provided the server must automatically select an available port.
```
